augmentedreality/Common/source/Android/ArCoreWrapper.cpp

```diff
@@ -296,24 +285,31 @@ void ArCoreWrapper::createArSession()
   if (m_arSession)
     return;
 
+  // request camera permission
+  auto checkPermissionResultFuture = QtAndroidPrivate::checkPermission(QtAndroidPrivate::PermissionType::Camera);
+  checkPermissionResultFuture.waitForFinished();
```

Comment:
~~Presumably this is ok to block like this? I assume this happens on the main thread.~~

Nevermind, I see you were calling `requestPermissionsSync` before.